src/PusherCrypto.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+namespace Pusher;
+
+class PusherCrypto
+{
+    private $encryption_key = '';
+
+    // The prefix any e2e channel must have
+    const ENCRYPTED_PREFIX = 'private-encrypted-';
+
+    // The exact length the user specified key must be
+    const SECRET_KEY_LENGTH = 32;
```

Comment:
True that. I'm thinking about whether we even need a key size limit like this which is why. Since it's just used for key derivation and it goes into a hashing function anyway its length isn't important (at least, lengths other than 32 characters will not break things!) anymore. What do you think?

Reply:
Since we are using the specified key to derive a key which is then used for encryption, the key the user specifies can be of any length (technically speaking a key of no length would work!). This limit was just carried over from the first implementation we did which was shared secret with no key derivation/distribution. I guess we could enforce a limit of greater than 10 chars or something?